fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -458,25 +458,38 @@ void FixedLagSmoother::transactionCallback(
       {
         started_ = true;
         ignited_ = true;
-        start_time = min_time;
-        setStartTime(min_time);
+        start_time = position->minStamp();
+        setStartTime(start_time);
+
+        // And purge out old transactions
+        //  - Either before or exactly at the start time
+        //  - Or with a minimum time before the minimum time of this ignition sensor transaction
+        //
+        // TODO(efernandez) Do '&min_time = std::as_const(start_ime)' when C++17 is supported and we can use
+        //                  std::as_const: https://en.cppreference.com/w/cpp/utility/as_const
+        pending_transactions_.erase(
```

Comment:
This performs the full interval check, so all other transactions `[min, max]` intervals are after the **ignition** sensor transaction interval.

Equality is allowed for the `min`, but not for the `max`. This is because we don't want a transaction that happened at the same time as the ignition sensor transaction. See the motivation and use case in this PR description.

Reply:
Note that this purge is only performed once, when the first ignition transaction is received. Because of the asynchronous nature of everything, it is possible that your initial graph transaction, `G1`, is received before the accompanying transaction, `T1`. In such a case, this purge code will check will fire _and then T1 is received_. Since `T1.maxStamp() == start_time`, the check above `if (started_ && max_time < start_time)` will also not purge out the `T1` transaction. I feel like those two checks should both use `<= start_time`